base/iobuffer.jl

```diff
@@ -30,6 +30,35 @@ end
 # allocate Vector{UInt8}s for IOBuffer storage that can efficiently become Strings
 StringVector(n::Integer) = unsafe_wrap(Vector{UInt8}, _string_n(n))
 
+"""
+    ensure_initialized!(buffer::GenericIOBuffer)
+
+If needed, re-initialize/allocate the underlying `data` array of `buffer`,
+which may be needed after `take!(buffer)` is called.  This function
+is normally called automatically by functions that read/write
+the buffer, but you may need to call it manually if you are accessing
+`buffer.data` directly and `take!` may have been called.
+"""
+@inline function ensure_initialized!(buf::GenericIOBuffer)
+    if buf.size < 0
```

Comment:
Maybe it is worth to only inline the fast path?

Reply:
Done.